daemon/config/config.go

```diff
@@ -351,8 +352,21 @@ func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Con
 	}
 
 	reader = bytes.NewReader(b)
-	err = json.NewDecoder(reader).Decode(&config)
-	return &config, err
+	if err := json.NewDecoder(reader).Decode(&config); err != nil {
```

Comment:
I think adding the flag check (in aeb461ed64758ebbccd0be303ae5a10cdfcc70dd)  is fine without adding more config validation.